### PR TITLE
Sage/<deploying celo dapp to celo network with web3py>

### PR DESCRIPTION
**Pull request by yusuf1990**
My tutorial follows the [Tutorial Contribution Guidelines](https://docs.celo.org/community/celo-sage/contribution-guidelines).
 The title of the Pull Request matches the title of my tutorial.
 I've followed the tutorial template structure.
 I've proofread the tutorial for spelling and grammar mistakes.
 I've thoroughly explained the code, with inline code comments and text explanations.
 I've run a linter & prettifier to format my code blocks.
 I've run the tutorial to test it and the code works.

### CODE REVIEW

File: blog/01-05-2023-deploying-celo-dapp-to-celo-network-with-web3py/deploying-celo-dapp-to-celo-network-with-web3py.md [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+```
+
+
+**Now, let's explain the code one by one**
+
+First, we will install the `web3.py` library. To install `web3.py`, you can run the following code in your terminal or command prompt or any Python IDE. I am using google colab here
+
+```py
+	pip install web3
+```
```
Comment (yafiabiyyu) [inline blog/01-05-2023-deploying-celo-dapp-to-celo-network-with-web3py/deploying-celo-dapp-to-celo-network-with-web3py.md:90]:
```suggestion
```python
pip install web3
```
```

File: blog/01-05-2023-deploying-celo-dapp-to-celo-network-with-web3py/deploying-celo-dapp-to-celo-network-with-web3py.md [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+
+Next, we'll now set up the web3 connection to the celo network. Here's an example code for setting up a web3 connection to the Celo network using `web3.py`:
+
+```py
+	 #First import the necessary library
+	 from web3 import Web3, HTTPProvider   
+         w3 = Web3(Web3.HTTPProvider('https://forno.celo.org'))
+```
```
Comment (yafiabiyyu) [inline blog/01-05-2023-deploying-celo-dapp-to-celo-network-with-web3py/deploying-celo-dapp-to-celo-network-with-web3py.md:98]:
```suggestion
```python
# First import the necessary library

from web3 import Web3, HTTPProvider
from web3.contract import ConciseContract 

w3 = Web3(Web3.HTTPProvider('https://forno.celo.org'))
```
```

File: blog/01-05-2023-deploying-celo-dapp-to-celo-network-with-web3py/deploying-celo-dapp-to-celo-network-with-web3py.md [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+
+##  Prerequisites
+1. Fundamental understanding of programming principles
+2. Knowledge of deploying and hosting applications
+3. Familiarity with blockchain technology
+
+##  Requirements
+1. Web3.py (Python library that allow you to interact with the blockchain)
+2. Python IDE (To run the code)
+3. Celo Network (To interact with the celo blockchain using web3.py, you need to connect to a celo network)
```
Comment (yafiabiyyu) [inline blog/01-05-2023-deploying-celo-dapp-to-celo-network-with-web3py/deploying-celo-dapp-to-celo-network-with-web3py.md:28]: Create more specific

File: blog/01-05-2023-deploying-celo-dapp-to-celo-network-with-web3py/deploying-celo-dapp-to-celo-network-with-web3py.md [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+
+# Create a concise contract instance
+from web3.contract import ConciseContract
+MyContract = concise_contract(w3.eth.contract(address=contract_address, abi=abi))
+
+# Example function call on the deployed contract
+result = MyContract.my_function()
+print(result)
+
+```
```
Comment (yafiabiyyu) [inline blog/01-05-2023-deploying-celo-dapp-to-celo-network-with-web3py/deploying-celo-dapp-to-celo-network-with-web3py.md:81]:
```suggestion
##  Here is the overview of our code

```python
from web3 import Web3, HTTPProvider
from web3.contract import ConciseContract

# Set up web3 connection to Celo network
w3 = Web3(Web3.HTTPProvider('https://forno.celo.org'))#Web3(HTTPProvider('https://forno.celo.org'))

# Replace with your contract bytecode and ABI
bytecode ='0x6080604052600080553.....'

abi = []

# Replace with your Celo account private key and address
private_key = private_key
address = your_address

# Create a contract instance
MyContract = w3.eth.contract(abi=abi, bytecode=bytecode)

# Estimate the gas cost for deploying the contract
gas_estimate = w3.eth.estimate_gas({'from': address, 'data': bytecode})

# Build the transaction
tx = {
    'from': address,
    'gas': gas_estimate,
    'gasPrice': Web3.to_wei('10', 'gwei'),
    'nonce': w3.eth.get_transaction_count(address),
    'data': bytecode
}

# Sign the transaction
signed_tx = w3.eth.account.sign_transaction(tx, private_key)

# Send the transaction
tx_hash = w3.eth.send_raw_transaction(signed_tx.rawTransaction)

# Wait for the transaction to be mined
tx_receipt = w3.eth.wait_for_transaction_receipt(tx_hash)

# Get the deployed contract address
contract_address = tx_receipt.contract_address

# Create a concise contract instance
MyContract = concise_contract(w3.eth.contract(address=contract_address, abi=abi))

# Example function call on the deployed contract
result = MyContract.my_function()
print(result)
```
```

Comment (yafiabiyyu) [reply blog/01-05-2023-deploying-celo-dapp-to-celo-network-with-web3py/deploying-celo-dapp-to-celo-network-with-web3py.md:81]: For bytecode and ABI it's better to make it shorter, so it's not too confusing

File: blog/01-05-2023-deploying-celo-dapp-to-celo-network-with-web3py/deploying-celo-dapp-to-celo-network-with-web3py.md (hunk 1 above)
Comment (yafiabiyyu) [reply blog/01-05-2023-deploying-celo-dapp-to-celo-network-with-web3py/deploying-celo-dapp-to-celo-network-with-web3py.md:90]: Remove unneeded spaces

File: blog/01-05-2023-deploying-celo-dapp-to-celo-network-with-web3py/deploying-celo-dapp-to-celo-network-with-web3py.md [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+
+```py
+bytecode ='0x60806040526000805534801561001457600080fd5b50610159806100246000396000f3fe608060405234801561001057600080fd5b50600436106100465760003560e01c806360fe47b11461004b5780636d4ce63c1461009a5780638da5cb5b146100d4575b600080fd5b6100536100e2565b6040518082815260200191505060405180910390f35b61006f6100e8565b6040518082815260200191505060405180910390f35b61008d6100f0565b6040518082815260200191505060405180910390f35b6000809054906101000a900460ff1681565b6000600160009054906101000a900473ffffffffffffffffffffffffffffffffffffffff1673ffffffffffffffffffffffffffffffffffffffff1663a9059cbb6040518163ffffffff1660e01b8152600401808373ffffffffffffffffffffffffffffffffffffffff1673ffffffffffffffffffffffffffffffffffffffff1681526020018281526020019250505060405180910390a15056fea2646970667358221220b3aa7e3faa5eb5ec5b5aeb81cf7ea317e752f969ce92769d9f9dc3f3a1422f2564736f6c634300060b0033' #'0xc2EdaD668740f1aA35E4D8f227fB8E17dcA888Cd'
+
+abi = [{'constant': True, 'inputs': [], 'name': 'myFunction', 'outputs': [{'name': '', 'type': 'uint256'}], 'payable': False, 'stateMutability': 'view', 'type': 'function'}]
+MyContract = w3.eth.contract(abi=ABI, bytecode=bytecode)
+```
```
Comment (yafiabiyyu) [inline blog/01-05-2023-deploying-celo-dapp-to-celo-network-with-web3py/deploying-celo-dapp-to-celo-network-with-web3py.md:109]:
```suggestion
```python
bytecode ='0x6080604052600080553.....'

abi = []

MyContract = w3.eth.contract(abi=ABI, bytecode=bytecode)
```
```

File: blog/01-05-2023-deploying-celo-dapp-to-celo-network-with-web3py/deploying-celo-dapp-to-celo-network-with-web3py.md [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+
+1. Go to [the celo explorer page](https://explorer.celo.org)
+2. Search for your contract on celo explorer and copy the bytecode from the "Bytecode" section of the contract details.
+
+
+Let's estimate the gas cost for deploying the contract with following code
+
+``` py
+gas_estimate = w3.eth.estimate_gas({'from': address, 'data': bytecode})
+```
```
Comment (yafiabiyyu) [inline blog/01-05-2023-deploying-celo-dapp-to-celo-network-with-web3py/deploying-celo-dapp-to-celo-network-with-web3py.md:128]:
```suggestion
``` python
gas_estimate = w3.eth.estimate_gas({'from': address, 'data': bytecode})
```
```

File: blog/01-05-2023-deploying-celo-dapp-to-celo-network-with-web3py/deploying-celo-dapp-to-celo-network-with-web3py.md [hunk 7]
```diff
@@ -0,0 +1,212 @@
...
+Now we can make the transaction, first step is building the transaction. Here's the code for building the transaction to deploy the contract:
+
+```py
+tx = {
+ 'from': address,
+ 'gas': gas_estimate,
+ 'gasPrice': Web3.to_wei('10', 'gwei'), 
+'nonce': w3.eth.get_transaction_count(address),
+ 'data': bytecode 
+}
```
Comment (yafiabiyyu) [inline blog/01-05-2023-deploying-celo-dapp-to-celo-network-with-web3py/deploying-celo-dapp-to-celo-network-with-web3py.md:142]:
```suggestion
```python
tx = {
    'from': address,
    'gas': gas_estimate,
    'gasPrice': Web3.to_wei('10', 'gwei'), 
    'nonce': w3.eth.get_transaction_count(address),
    'data': bytecode 
}
```

File: blog/01-05-2023-deploying-celo-dapp-to-celo-network-with-web3py/deploying-celo-dapp-to-celo-network-with-web3py.md [hunk 8]
```diff
@@ -0,0 +1,212 @@
...
+```py
+pip install web3-contract
+```
+Now, we can generate the contract instance
+
+```py
+#First import the ConciseContract
+from web3.contract import ConciseContract
+MyContract = ConciseContract(w3.eth.contract(address=contract_address, abi=abi))
+```
```
Comment (yafiabiyyu) [inline blog/01-05-2023-deploying-celo-dapp-to-celo-network-with-web3py/deploying-celo-dapp-to-celo-network-with-web3py.md:181]: Move the installation process at the beginning and also the import process at the beginning

File: blog/01-05-2023-deploying-celo-dapp-to-celo-network-with-web3py/deploying-celo-dapp-to-celo-network-with-web3py.md [hunk 9]
```diff
@@ -0,0 +1,212 @@
...
+
+
+
+
+
+
+
+
+
+
```
Comment (yafiabiyyu) [inline blog/01-05-2023-deploying-celo-dapp-to-celo-network-with-web3py/deploying-celo-dapp-to-celo-network-with-web3py.md:212]: Remove unnecessary spaces

**Review by Ishanpathak1 — APPROVED** [review]
Approved @developerkunal

**Pull request merged**